elastic/apm-server
# translate otel messaging.* to ecs

## Motivation/summary

Translate relevant otel messaging.* semantic conventions to their
elastic-equivalent fields.

closes https://github.com/elastic/apm-server/issues/5094

## Checklist




## How to test these changes

Ingest messaging spans from an otel-instrumented messaging service.

EDIT: https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/instrumentation/github.com/Shopify/sarama/otelsarama/example

Check the transactions and spans in the kibana UI for the following:

transaction:
- `transaction.type=messaging`
- Message metadata has fields `queue.name`, `system`, and `operation`.

span:
- `span.type=messaging`
- `span.subtype` is the framework name, eg. kafka
- `span.action` is one of `receive`, `send`, or `process`
- Message metadata has fields `queue.name`.
